include/zephyr/bindesc.h

```diff
@@ -0,0 +1,405 @@
+ * Do not change without syncing the definitions in both files!
+ */
+
+/** The app version string such as "1.2.3" */
+#define BINDESC_ID_APP_VERSION_STRING 0x800
+
+/** The app version major such as 1 */
+#define BINDESC_ID_APP_VERSION_MAJOR 0x801
+
+/** The app version minor such as 2 */
+#define BINDESC_ID_APP_VERSION_MINOR 0x802
+
+/** The app version patchlevel such as 3 */
+#define BINDESC_ID_APP_VERSION_PATCHLEVEL 0x803
+
+/** The app version number such as 0x10203 */
+#define BINDESC_ID_APP_VERSION_NUMBER 0x804
+
+/** The kernel version string such as "3.4.0" */
+#define BINDESC_ID_KERNEL_VERSION_STRING 0x900
+
+/** The kernel version major such as 3 */
+#define BINDESC_ID_KERNEL_VERSION_MAJOR 0x901
+
+/** The kernel version minor such as 4 */
+#define BINDESC_ID_KERNEL_VERSION_MINOR 0x902
+
+/** The kernel version patchlevel such as 0 */
+#define BINDESC_ID_KERNEL_VERSION_PATCHLEVEL 0x903
+
+/** The kernel version number such as 0x30400 */
+#define BINDESC_ID_KERNEL_VERSION_NUMBER 0x904
+
+/** The year the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_YEAR 0xa00
+
+/** The month of the year the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_MONTH 0xa01
+
+/** The day of the month the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_DAY 0xa02
+
+/** The hour of the day the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_HOUR 0xa03
+
+/** The minute the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_MINUTE 0xa04
+
+/** The second the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_SECOND 0xa05
+
+/** The UNIX time (seconds since midnight of 1970/01/01) the image was compiled in */
+#define BINDESC_ID_BUILD_TIME_UNIX 0xa06
+
+/** The date and time of compilation such as "2023/02/05 00:07:04" */
+#define BINDESC_ID_BUILD_DATE_TIME_STRING 0xa07
+
+/** The date of compilation such as "2023/02/05" */
+#define BINDESC_ID_BUILD_DATE_STRING 0xa08
+
+/** The time of compilation such as "00:07:04" */
+#define BINDESC_ID_BUILD_TIME_STRING 0xa09
+
+/** The name of the host that compiled the image */
+#define BINDESC_ID_HOST_NAME 0xb00
+
+/** The C compiler name */
+#define BINDESC_ID_C_COMPILER_NAME 0xb01
+
+/** The C compiler version */
+#define BINDESC_ID_C_COMPILER_VERSION 0xb02
+
+/** The C++ compiler name */
+#define BINDESC_ID_CXX_COMPILER_NAME 0xb03
+
+/** The C++ compiler version */
+#define BINDESC_ID_CXX_COMPILER_VERSION 0xb04
+
+#define BINDESC_TAG_DESCRIPTORS_END BINDESC_TAG(DESCRIPTORS_END, 0x0fff)
+
+/**
+ * @cond INTERNAL_HIDDEN
+ */
+
+/*
+ * Utility macro to generate a tag from a type and an ID
+ *
+ * type - Type of the descriptor, UINT, STR or BYTES
+ * id - Unique ID for the descriptor, must fit in 12 bits
+ */
+#define BINDESC_TAG(type, id) ((BINDESC_TYPE_##type & 0xf) << 12 | (id & 0x0fff))
+
+/*
+ * Utility macro to get the name of a bindesc entry
+ */
+#define BINDESC_NAME(name) bindesc_entry_##name
+
+/*
+ * Utility macro to convert a uint32_t to a uint8_t array
```

Comment:
I think you have all you need here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/byteorder.h

No need to add more macros?